1-Draft/RFCNNNN-AdditionalTelemetry.md

```diff
@@ -0,0 +1,175 @@
+
+## Specification
+
+Add telemetry to track the following metrics:
+
+- Count of unique devices (PowerShell)
+- Count of unique users (PowerShell)
+- Count of execution types (ex. cmdlets, native binaries/applications) (PowerShell)
+- Count of session types (ex. hosted vs not hosted sessions) (PowerShell)
+- Hosting application of the of the PowerShell session (ex. VSCode, ISE, AZDevOps) (PowerShell)
+- Version of PowerShell, PowerShellGet, PackageManagement, Nuget and platform
+  used to install from PSGallery (PowerShellGet)
+
+### Other features
+
+- Public PowerBi DashBoard to make the aggregated statistics available to the community.
+  Find a mocked-up PowerBi DashBoard with randomly generated data at the end of the document.
+  It includes the following reports:
+  - Unique user trends
+  - Unique devices trends
+  - Types of user &mdash;based on types of executions taking place
+  - VSCode usage trends
+  - Application trends
+  - Hosted Scenario Trends
+  - PowerShellGet installation trends
+- A cmdlet for providing feedback (Send-PSFeedback):
+  - This cmdlet would take in a string and send it back to the PowerShell team
+  - The cmdlet will also send Platform data (OS/version) and Shell data ($PSVersionTable)
+  - The cmdlet will have an -OpenGitHubIssue switch that will allow the user to open a GitHubIssue
+    in the PowerShell/PowerShell repository
+  - The user will have the option to include the the last error using an -IncludeError switch
+  - If the command is not successful (ex. there is no Web Connection) the cmdlet will return a
+    warning and alternative instructions for support through GitHub
+  - If no location switch is used the cmdlet will default send the feedback to the PowerShell
+    User Voice
+  - Based on usage of this cmdlet there would potential for additional parameters
+- Disabling, and re-enabling telemetry will be available through a new cmdlet (Set-PSTelemetry):
+  - The cmdlet will stop all collection of telemetry for the remainder of that session and stop
+    telemetry from being collected on subsequent starts
+  - The user can request their telemetry is deleted by using a Remove-PSTelemetry cmdlet
+  - In order to have their telemetry deleted the user will be required to provide their System GUID
+    and User GUID
+  - If the cmdlet is unable to send the information the cmdlet will return a warning and alternative
+    instructions for support through GitHub. It will be specified that GUIDs are unique identifiers
+    and should not be shared publicly on GitHub.
+- There will be three options for the telemetry environment variable which can all be set manually
+  or using the Set-PSTelemetry cmdlet
+  - $True indicates all telemetry will be collected
+  - $False indicates no telemetry will be collected
+  - $NonIdentifying indicates that only non-identifying telemetry will be collected i.e. no GUIDs
+    will be collected

```

Comment:
What will be the default value if it is _not_ set?

Reply:
It should be **stressed** that non-identifying telemetry should be the default. Especially when the stated privacy goal is:

> Respecting the privacy of our users has been the principal factor in determining which metrics we plan to collect.

Reply:
If it's opt-out and not opt-in, is that GDPR compliant?

Reply:
I very much believe the default should be $false and opt-in only. I'll happily opt-in my home machines, but I can't opt-in at work due to company policy so it would be highly annoying if $true was the default ($NonIdentifying *might* be ok). Not to mention it being incredibly annoying if I have to opt out any container I want to run with minimum specifications which is basically anything in a cloud provider (Azure/AWS/whatever).

Reply:
I think that I caused some confusion in naming one of the switch settings to be "$NonIdentifying" in truth none of the metrics are identifying because GUIDs are **not** hashes that can be traced back to a person. I am thinking it makes the most sense to just have the options as simply $True and $False. In terms of company policy since these are enviromental variables they should be able to be set using Group Policy.

Reply:
@SydneyhSmith Thanks for the reply. I feel if I don't ask it someone else will:

What will the default option be?

Reply:
The default would be $True